lib/reek/configuration/configuration_validator.rb

```diff
@@ -10,26 +10,14 @@ module ConfigurationValidator
 
       # :reek:UtilityFunction
       def smell_type?(key)
-        case key
-        when Class
-          true
-        when String
-          begin
-            Reek::SmellDetectors.const_defined? key
-          rescue NameError
-            false
-          end
-        end
+        Reek::SmellDetectors.const_defined? key
+      rescue NameError
```

Comment:
If `key` is now expected to be a string or a symbol, is it still necessary to account for references to uninitialized constants? 🤔 Would make more sense to me if the `false` was something like exiting with a deprecation message.

(I'm not sure what this did in the original implementation, since `key` was already known to be a string.)

Reply:
Key may have a form like `lib/foo.rb` in this case so a NameError could still be thrown.

Reply:
Thanks for clarifying! That didn't cross my mind. 🙂

Reply:
It didn't cross mine either, so I removed it and the specs failed :slightly_smiling_face:.